### PR TITLE
th-15: Websockets integration

### PR DESCRIPTION
**Pull request by DmytroVoronkov**
- Created socket service in the backend 
- Created socket class in the client 
- Created shared socket enums 
- Created socket status on root page
image

### CODE REVIEW

File: backend/src/libs/packages/server-application/server-app.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ import { type IConfig } from '~/libs/packages/config/config.js';
 import { type IDatabase } from '~/libs/packages/database/database.js';
 import { HttpCode, HttpError } from '~/libs/packages/http/http.js';
 import { type ILogger } from '~/libs/packages/logger/logger.js';
+import { socket } from '~/libs/packages/socket/socket.js';
```
Comment (kalinkaaaa14) [inline backend/src/libs/packages/server-application/server-app.ts]:
```suggestion
import { socket as socketService } from '~/libs/packages/socket/socket.js';
```

File: backend/src/libs/packages/socket/socket.service.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  private io: SocketServer | null = null;
+
+  public getIo(): SocketServer {
+    return this.io as SocketServer;
+  }
+
+  public initializeIo(app: FastifyInstance): void {
+    const io = new SocketServer(app.server, {
+      cors: { origin: '*' },
+    });
```
Comment (kalinkaaaa14) [inline backend/src/libs/packages/socket/socket.service.ts]:
```suggestion
    this.io = new SocketServer(app.server, {
      cors: { origin: '*' },
    });
```

File: shared/src/libs/packages/socket/libs/enums/socket-namespaces.enum.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+const SocketNamespaces = {
```
Comment (kalinkaaaa14) [inline shared/src/libs/packages/socket/libs/enums/socket-namespaces.enum.ts]: enum should be in singular form

File: shared/src/libs/packages/socket/libs/enums/server-socket-events.enum.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+const ServerSocketEvents = {
```
Comment (kalinkaaaa14) [inline shared/src/libs/packages/socket/libs/enums/server-socket-events.enum.ts]: enum should be in singular form

File: shared/src/libs/packages/socket/libs/enums/client-socket-events.enum.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+const ClientSocketEvents = {
```
Comment (kalinkaaaa14) [inline shared/src/libs/packages/socket/libs/enums/client-socket-events.enum.ts]: enum should be in singular form

File: frontend/package.json [hunk 6]
```diff
@@ -21,7 +21,9 @@
     "react-dom": "18.2.0",
     "react-hook-form": "7.43.5",
     "react-redux": "8.1.2",
-    "react-router-dom": "6.8.2"
+    "react-router-dom": "6.8.2",
+    "socket.io-client": "4.7.2",
+    "stylelint": "15.10.2"
```
Comment (kalinkaaaa14) [inline frontend/package.json]: stylelint is in root package json

File: frontend/src/libs/packages/socket/socket-api.ts
Comment (kalinkaaaa14) [inline frontend/src/libs/packages/socket/socket-api.ts:1]: rename to socket.package.ts

**Review by kalinkaaaa14 — DISMISSED** [review]

File: frontend/src/libs/packages/socket/socket-api.ts [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import { type Socket as BaseSocket, io } from 'socket.io-client';
+
+class Socket {
+  public getInstance(): BaseSocket {
+    return io('http://localhost:3001');
```
Comment (Druxight) [inline frontend/src/libs/packages/socket/socket-api.ts]: move to the .env file

**Review by Druxight — CHANGES_REQUESTED** [review]

**Pull request closed without merging**